src/components/Dashboard/Shop/Product.js

```diff
@@ -37,31 +38,34 @@ export class Product extends Component {
     return this.setState({ error: null });
   };
 
-  calculateTotalPrice = (pcsOrder, price) => pcsOrder * price;
-
   showAddedToCart = () => {
     const { toggleAddedToCart } = this.state;
     this.setState({ toggleAddedToCart: !toggleAddedToCart });
   };
 
-  moreThanInStock = () => (!!this.state.error);
+  moreThanInStock = () => !!this.state.error;
 
   redirectToCart = () => this.props.history.push('/cart');
 
   handleClick = (product) => {
     const {
-      addProduct, cart, updateProduct, token,
+      addProduct,
+      cart: { productsInCart, list, byId },
```

Comment:
Passing the raw store structure to the component complicates it's logic. And if multiple components use the same data, then logic for transforming it will be replicated in all of them.
That's why selectors should pass only the data that component needs. In this case

I see there is quite a few strange things going on here, and we'll need to discuss them live.